modules/runner/src/testFixtures/java/org/apache/ignite/internal/ClusterPerClassIntegrationTest.java

```diff
@@ -386,38 +384,37 @@ private static List<List<Object>> executeAwaitingIndexes(IgniteImpl node, Functi
         CatalogManager catalogManager = node.catalogManager();
 
         // Get existing indexes
-        Set<Integer> existing = catalogManager.indexes(catalogManager.latestCatalogVersion())
-                .stream().map(CatalogObjectDescriptor::id)
+        Set<Integer> existing = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
+                .map(CatalogObjectDescriptor::id)
                 .collect(Collectors.toSet());
 
         List<List<Object>> result = statement.apply(node);
 
         // Get indexes after a statement and compute the difference
-        Set<Integer> difference = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
+        List<Integer> difference = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
                 .map(CatalogObjectDescriptor::id)
-                .collect(Collectors.toSet());
-
-        difference.removeAll(existing);
+                .filter(id -> !existing.contains(id))
+                .collect(Collectors.toList());
 
         if (difference.isEmpty()) {
             return result;
         }
 
         // If there are new indexes, wait for them to become available.
+        HybridClock clock = node.clock();
 
         try {
-            assertTrue(waitForCondition(() -> {
-                int latestVersion = catalogManager.latestCatalogVersion();
-                int notAvailable = 0;
-
-                for (CatalogIndexDescriptor index : catalogManager.indexes(latestVersion)) {
-                    if (!index.available() && difference.contains(index.id())) {
-                        notAvailable++;
-                    }
-                }
-
-                return notAvailable == 0;
-            }, 10_000));
+            assertTrue(waitForCondition(
+                    () -> {
+                        long now = clock.nowLong();
+
+                        return difference.stream()
+                                .map(id -> catalogManager.index(id, now))
+                                .allMatch(indexDescriptor -> indexDescriptor != null && indexDescriptor.available());
+                    },
+                    10,
```

Comment:
I copy-pasted it from a similar method in this class...

Reply:
copy-past - it's bad explanation.
We must do it consciously

Reply:
previous approach here looks better

Reply:
It would look inconsistent, that the other method that also waits for indices uses a different timing

Reply:
Why other method waits so long? As I see you was single reviewer for that change )

Reply:
Why is that a big deal? I don't think 10 seconds is better than 30 seconds in this context

Reply:
Could you imaging real situatio when index will became available for so long time for our tests? Why we don't use 10 hours here in case it's not a problem?
My point is the following: in case our code have bugs all tests will waiting long time and significantly increasing time of whole build. We should use values that make sense.

Reply:
Changed to 10 seconds